vulkaninfo/vulkaninfo.h

```diff
@@ -2026,13 +2000,17 @@ struct hash<PropFlags> {
 };
 }  // namespace std
 
+const std::vector<FormatRange> &get_format_ranges();
+
 // Used to sort the formats into buckets by their properties.
-std::unordered_map<PropFlags, std::vector<VkFormat>> FormatPropMap(AppGpu &gpu) {
-    std::unordered_map<PropFlags, std::vector<VkFormat>> map;
-    for (auto fmtRange : gpu.supported_format_ranges) {
-        for (int32_t fmt = fmtRange.first_format; fmt <= fmtRange.last_format; ++fmt) {
-            PropFlags pf = get_format_properties(gpu, static_cast<VkFormat>(fmt));
-            map[pf].push_back(static_cast<VkFormat>(fmt));
+std::unordered_map<PropFlags, std::set<VkFormat>> FormatPropMap(AppGpu &gpu) {
```

Comment:
Can this be passed by const ref?

Reply:
No cause its generated by the function. That would inhibit GRVO.